src/compiler/binder.ts

```diff
@@ -2503,8 +2503,27 @@ namespace ts {
             }
         }
 
+        function needParameterStrictModeCheck(node: Node) {
+            if (node.parent.kind === SyntaxKind.ConstructorType || node.parent.kind === SyntaxKind.FunctionType) {
+                return false;
+            }
+
+            if (node.parent.parent && (
+                node.parent.parent.kind === SyntaxKind.InterfaceDeclaration ||
+                node.parent.parent.kind === SyntaxKind.TypeLiteral
+            )) {
+                return false;
+            }
+
+            if (isDeclarationFile(file) || isInAmbientContext(node)) {

```

Comment:
I hope so but it only checks whether parents node has `declare` modifiers. I first tried modifying it but it broke other tests.